### PR TITLE
Fix for #49 (prerelease test result failure)

### PR DESCRIPTION
**Pull request by 130s**
<s>**DO NOT Review nor Merge yet.**</s>
Ready to be reviewed.

### CODE REVIEW

**Comment by 130s** [comment]
Ready to be reviewed. @ipa-mdl Could you take a look? This solves #49.

File: travis.sh [hunk 1]
```diff
@@ -170,11 +170,12 @@ travis_time_end  # rosdep_install
...
-if [ "$PRERELEASE" == true ] && [ -e ${CI_SOURCE_PATH}/$CI_PARENT_DIR/ros_pre-release.sh ]; then 
-  ${CI_SOURCE_PATH}/$CI_PARENT_DIR/ros_pre-release.sh
-  catkin_test_results build && (echo 'ROS Prerelease Test went successful.'; exit 0) || error
+if [ "$PRERELEASE" == true ] && [ -e ${CI_SOURCE_PATH}/$CI_PARENT_DIR/ros_pre-release.sh ]; then
+  ${CI_SOURCE_PATH}/${CI_PARENT_DIR}/ros_pre-release.sh
+  retval_prerelease=$?
+  if [ $retval_prerelease -eq 0 ]; then success; else error; fi  # Internally called travis_time_end for prerelease_from_travis_sh
+  # With Prerelease option, we want to stop here without running the rest of the code.
 fi
-travis_time_end  # prerelease_from_travis_sh
```
Comment (ipa-mdl) [inline travis.sh]: did you intend to delete this line? Should the start time (lin2 172) be deleted as well?

File: .travis.yml [hunk 2]
```diff
@@ -8,8 +8,10 @@ dist: trusty
 language: generic
 compiler:
   - gcc
-#notifications:
-#  Whoever receives success and/or failure via email
+notifications:
+  email:
+    recipients:
+      - gm130s@gmail.com
```
Comment (ipa-mdl) [inline .travis.yml]: I would add a comment to change this line if copied.

File: travis.sh (hunk 1 above)
Comment (130s) [inline travis.sh]: Removing `travis_time_end` in this line is intended. I now let [`success` and `error` functions do the work internally](https://github.com/ros-industrial/industrial_ci/pull/50/files/bb704c7af752fa389e1e9e4a5ad990eab3b6812b#diff-a188fc26ef16ad3194c8d957e9e0cd3dR78).

Prior to this PR, `error` function already ran `travis_time_end`, but only when non-zero result occurred. `error` also run `exit 1`, which was called only with non-zero. So it was kind of redundant and incomplete. I hope this PR is cleaner and complete, in terms of fold and exit.

File: .travis.yml (hunk 2 above)
Comment (130s) [inline .travis.yml]: Sorry, I don't understand your comment. I reverted back the comment line in my updated commit. Is that what you meant?

File: ros_pre-release.sh [hunk 3]
```diff
@@ -71,9 +72,10 @@ travis_time_start run_prerelease
 travis_time_end  # run_prerelease
 
 travis_time_start show_testresult
-catkin_test_results --verbose && (echo 'ROS Prerelease Test went successful.'; exit 0) || error
+catkin_test_results --verbose && { echo 'ROS Prerelease Test went successful.'; RETVAL=0; success; } || { RETVAL=1; error; }
 travis_time_end  # show_testresult
```
Comment (ipa-mdl) [inline ros_pre-release.sh]: So `travis_time_end` should not be used if block is terminated by `success` and `error`?

Comment (130s) [inline ros_pre-release.sh]: Good point, and my answer is no. `return` [in this line](https://github.com/ros-industrial/industrial_ci/pull/50/files#diff-69501095267bf26a6434210905896aa4R81) is important for the parent script to be able to capture the result. So we don't want this script to exit at this line yet and thus don't want to call `success`.

Hope this variation is not unnecessarily complicated, but maybe it is. I appreciate your opinion.

File: .travis.yml (hunk 2 above)
Comment (ipa-mdl) [inline .travis.yml]: You have provided your mail address here, so adding a hint to change it for users that copied this config might prevent you from spam ;)

Comment (130s) [inline .travis.yml]: Very thoughtful of you :) Addressed.

**Comment by 130s** [comment]
@ipa-mdl thank you for taking a look many times. If you don't have more to say I'll go ahead merge.

**Comment by 130s** [comment]
Wait Prerelease Tests aren't passing. I'll look.

**Comment by 130s** [comment]
Ready for the review.

File: .travis.yml [hunk 4]
```diff
@@ -8,22 +8,27 @@ dist: trusty
 language: generic
 compiler:
   - gcc
-#notifications:
-#  Whoever receives success and/or failure via email
+notifications:
+  #  Whoever receives success and/or failure via email
+  email:
+    recipients:
+      - gm130s@gmail.com  # Change this to yours you copy this file.
```
Comment (ipa-mdl) [inline .travis.yml]: +if ? ;)

**Comment by ipa-mdl** [comment]
I think this PR is a litte bit mixed up, since it includes fixes and unrelated(?) enhancements, but the code it is much cleaner now, so +1 for merge :)

**Comment by 130s** [comment]
Thanks. Indeed the most of commits may seem unrelated but all of them contribute to the fix.
The last comment is addressed now.

**Pull request merged**